tsl/src/bgw_policy/retention_api.c

```diff
@@ -288,15 +277,34 @@ policy_retention_add(PG_FUNCTION_ARGS)
 }
 
 Datum
-policy_retention_remove(PG_FUNCTION_ARGS)
+policy_retention_add(PG_FUNCTION_ARGS)
 {
-	Oid table_oid = PG_GETARG_OID(0);
-	bool if_exists = PG_GETARG_BOOL(1);
-	Cache *hcache;
-	Hypertable *hypertable;
+	/* behave like a strict function */
+	if (PG_ARGISNULL(0) || PG_ARGISNULL(1) || PG_ARGISNULL(2))
+		PG_RETURN_NULL();
+
+	Oid ht_oid = PG_GETARG_OID(0);
+	Datum window_datum = PG_GETARG_DATUM(1);
+	bool if_not_exists = PG_GETARG_BOOL(2);
+	Oid window_type = PG_ARGISNULL(1) ? InvalidOid : get_fn_expr_argtype(fcinfo->flinfo, 1);
+	Interval default_schedule_interval =
+		PG_ARGISNULL(3) ? (Interval) DEFAULT_RETENTION_SCHEDULE_INTERVAL : *PG_GETARG_INTERVAL_P(3);
 
 	TS_PREVENT_FUNC_IF_READ_ONLY();
 
+	return policy_retention_add_internal(ht_oid,
+										 window_type,
+										 window_datum,
+										 default_schedule_interval,
+										 if_not_exists);
+}
+
+Datum
+policy_retention_remove_internal(Oid table_oid, bool if_exists)
+{
+	Cache *hcache;
```

Comment:
Nit: I usually add the internal functions after when I do this kind of refactoring because it makes the patch easier to review.